### PR TITLE
Add "Copy Search URL" plugin

### PR DESCRIPTION
**Pull request by czottmann**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/czottmann/obsidian-copy-search-url

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by liamcain** [comment]
- [this.button.parentNode?.removeChild(this.button);](https://github.com/czottmann/obsidian-copy-search-url/blob/d18ff31efa4c599c970c338b8aefed1a7db22126/main.ts#L54) You might want to add a check to make sure that `this.button` isn't null. Also you can use [`.detach()`](https://github.com/obsidianmd/obsidian-api/blob/1b4f6e2e5753c8ac2d538bc5d55c38f228450cb2/obsidian.d.ts#L38) which is available on all HTMLElements in Obsidian. So this would just become `this.button?.detach();`

- [return (this.getSearchLeaf().view as SearchLeafView)?.getQuery();](https://github.com/czottmann/obsidian-copy-search-url/blob/d18ff31efa4c599c970c338b8aefed1a7db22126/main.ts#L66-L67) You might want to return `""` if the search leaf can't be found, just so that your return type is always a string. It shouldn't make a difference, but you are checking for emptiness inside the click event listener so it might be safer.

**Comment by czottmann** [comment]
Thanks, @liamcain, much appreciated!
